lnwallet/channel.go

```diff
@@ -1245,7 +1292,12 @@ func (lc *LightningChannel) ExtendRevocationWindow() (*lnwire.CommitRevocation,
 // should be called when preparing to send an outgoing HTLC.
 // TODO(roasbeef): check for duplicates below? edge case during restart w/ HTLC
 // persistence
-func (lc *LightningChannel) AddHTLC(htlc *lnwire.HTLCAddRequest) uint32 {
+func (lc *LightningChannel) AddHTLC(htlc *lnwire.HTLCAddRequest) (uint32, error) {
+	if lc.estimateCommitTxCost(lc.theirLogCounter, lc.ourLogCounter,
```

Comment:
There should also be an identical check in `ReceiveHTLC` . The spec dictates that either side should **never** send an HTLC log modification that would result in an over weight commitment transaction if a signature were sent directly afterwards. Additionally, as a defensive measure we should also include the check on HTLC receives. 

One could argue that the check above is enough, but we might as well catch the protocol error early. 

Similarly, I think we should also insert the same check when _creating_ a new commitment state ourselves. The rationale is that this check may possibly catch any future/current bugs in the HTLC log evaluation logic which might cause us to erroneously create a new commitment that's overweight.